specutils/spectra/xrayspectrum1d.py

```diff
@@ -0,0 +1,542 @@
+    if unit_string in EV:
+        return u.eV
+    if unit_string in KEV:
+        return u.keV
+    if unit_string in ANGS:
+        return u.angstrom
+
+class XraySpectrum1D(Spectrum1D):
+    """
+    Spectrum properties specific to X-ray data
+
+    Parameters
+    ----------
+    bin_lo : numpy.ndarray
+        The left edges for bin values
+
+    bin_hi : numpy.ndarray
+        The right edges for bin values
+
+    bin_unit : string OR astropy.units.Unit
+        Unit for the bin values
+
+    counts : numpy.ndarray
+        Counts histogram for the X-ray spectrum
+
+    exposure : float
+        Exposure time for the dataset
+
+    arf : specutils.ARF or string, default None
+        Strings will be passed to ARF.__init__
+        All other input types are stored as arf attribute
+
+    rmf : specutils.RMF or string, default None
+        Strings will be passed to RMF.__init__
+        All other input types are stored as rmf attribute
+
+    rest_value : astropy.units.Quantity, default 0 Angstrom
+        See Spectrum1D rest_value input
+
+    Attributes
+    ----------
+    Inherits from Spectrum1D object.
+    The following inputs are stored as additional attributes.
+
+    bin_lo
+
+    bin_hi
+
+    counts
+
+    exposure
+
+    arf
+
+    rmf
+    """
+    def __init__(self, bin_lo, bin_hi, bin_unit, counts, exposure,
+                 arf=None, rmf=None, rest_value=0.0 * u.angstrom, **kwargs):
+        try:
+            axis_unit = u.Unit(bin_unit)
+        except:
+            axis_unit = _unit_parser(bin_unit)
+
+        bin_mid = 0.5 * (bin_lo + bin_hi) * axis_unit
+        Spectrum1D.__init__(self, spectral_axis=bin_mid, flux=counts,
+                            rest_value=rest_value, **kwargs)
+
+        self.bin_lo = bin_lo
+        self.bin_hi = bin_hi
+        self.exposure = exposure
+        self.assign_rmf(rmf)
+        self.assign_arf(arf)
+        return
+
+    # Convenience function for Xray people
+    @property
+    def counts(self):
+        return self.flux
+
+    def assign_arf(self, arf_inp):
+        """
+        Assign an ARF object to the XraySpectrum1D object
+
+        Input
+        -----
+        arf_inp : string
+            File name for the ARF (FITS file)
+
+        Returns
+        -------
+        Modifies the XraySpectrum1D.arf attribute
+        """
+        if isinstance(arf_inp, str):
+            self.arf = ARF(arf_inp)
+        else:
+            self.arf = arf_inp
+        return
+
+    def assign_rmf(self, rmf_inp):
+        """
+        Assign an RMF object to the XraySpectrum1D object
+
+        Input
+        -----
+        rmf_inp : string
+            File name for the RMF (FITS file)
+
+        Returns
+        -------
+        Modifies the XraySpectrum1D.rmf attribute
+        """
+        if isinstance(rmf_inp, str):
+            self.rmf = RMF(rmf_inp)
+        else:
+            self.rmf = rmf_inp
+        return
+
+    def apply_resp(self, mflux, exposure=None):
+        """
```

Comment:
apply_response is likely more clear than 'resp'

Reply:
Going with @eblur : calling is resp is common across the standard tools in X-ray astronomy. That does not mean that we need to keep it, but we should either consistently use what X-ray astronomy tools do, or consistently spell out everything.